core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/retry/RateLimitingTokenBucketTest.java

```diff
@@ -96,6 +96,17 @@ public void tryAcquireCapacity_capacitySufficient_returns0() {
         tb.setCurrentCapacity(5.0);
 
         assertThat(tb.tryAcquireCapacity(5.0)).isZero();
+        assertThat(tb.getCurrentCapacity()).isZero();
+    }
+
+    @Test
+    public void tryAcquireCapacity_amountGreaterThanCapacity_returnsNonZero() {
```

Comment:
Can you add a test where the `currentCapacity` is negative when `acquire()` is called as well?

Reply:
done